R/appMetadata.R

```diff
@@ -0,0 +1,378 @@
+appMetadata <- function(appDir,
+                        appFiles = NULL,
+                        appPrimaryDoc = NULL,
+                        quarto = NULL,
+                        contentCategory = NULL,
+                        isCloudServer = FALSE,
+                        metadata = list()) {
+
+  if (is.null(appFiles)) {
```

Comment:
Discovery of `appFiles` via `bundleFiles` feels like an operation that should be done by the caller, especially as its result is discarded.

There are separate calls to `appFiles` both here and in `inferAppMode`.

Reply:
I do it here because I think it's a reasonable default and it makes testing code much cleaner. I suspect this will end up looking cleaner as I continue to refactor.

Reply:
Actually, one way to eliminate an argument would be to combine `appDir` with `appFiles`/`appPrimaryDoc` to create absolute paths in `appMetadata()`. That would eliminate one argument from each of `inferQuartoInfo()`, `inferAppMode()`, `inferAppPrimaryDoc()`, `appHasParameters()`, and `detectPythonInDocuments()`.

Reply:
I'm OK keeping the `bundleFiles` call here, for now. Do we continue to need both calls?

If you make paths absolute, we'll need to make sure we retain the correct directory structure within the bundle.

Reply:
I'll think about this more in a follow PR.